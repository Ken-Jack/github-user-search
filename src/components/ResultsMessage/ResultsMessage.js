// @flow

import React from 'react';
import {connect} from 'react-fela';

type Props = {
  pageTotal: number,
  resultsTotal: number,
  searchTerm: string,
  styles: Object,
};

function ResultsMessage({searchTerm, resultsTotal, pageTotal, styles}: Props) {
  if (!searchTerm || !pageTotal) {return null;}

  return (
    <div className={styles.ResultsMessage_root}>
      <p className={styles.ResultsMessage_text}>
        Found {resultsTotal} {resultsTotal === 1 ? 'result' : 'results'} for <b>{searchTerm}</b>
      </p>
    </div>
  );
}

const styles = {
  ResultsMessage_root: () => ({
    display: 'flex',
    justifyContent: 'center',
  }),

  ResultsMessage_text: () => ({
    padding: 6,
    backgroundColor: '#fcf8bb',
    borderRadius: 8,
  }),
};

export default connect(styles)(ResultsMessage);
