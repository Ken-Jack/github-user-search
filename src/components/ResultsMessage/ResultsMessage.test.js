import React from 'react';
import felaSnapshot from 'test-util/fela-snapshot';

import ResultsMessage from './ResultsMessage';

describe('Component: ResultsMessage', () => {

  it('should render', () => {
    const component = felaSnapshot(
      <ResultsMessage
        searchTerm={'test'}
        pageTotal={10}
        resultsTotal={50}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should handle single results', () => {
    const component = felaSnapshot(
      <ResultsMessage
        searchTerm={'test'}
        pageTotal={1}
        resultsTotal={1}
      />
    );
    expect(component).toMatchSnapshot();
  });

});
