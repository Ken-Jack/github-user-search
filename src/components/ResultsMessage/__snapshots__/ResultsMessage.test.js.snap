// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ResultsMessage should handle single results 1`] = `
Object {
  "component": <div
    className="a b"
  >
    <p
      className="c d e"
    >
      Found 
      1
       
      result
       for 
      <b>
        test
      </b>
    </p>
  </div>,
  "styles": "
    .a{display:flex}
    .b{justify-content:center}
    .c{padding:6}
    .d{background-color:#fcf8bb}
    .e{border-radius:8}",
}
`;

exports[`Component: ResultsMessage should render 1`] = `
Object {
  "component": <div
    className="a b"
  >
    <p
      className="c d e"
    >
      Found 
      50
       
      results
       for 
      <b>
        test
      </b>
    </p>
  </div>,
  "styles": "
    .a{display:flex}
    .b{justify-content:center}
    .c{padding:6}
    .d{background-color:#fcf8bb}
    .e{border-radius:8}",
}
`;
